pkg/dns/dns.go

```diff
@@ -559,22 +559,38 @@ func getSkyMsg(ip string, port int) (*skymsg.Service, string) {
 //   6. The remaining segments match kd.domainPath.
 //   7. And federation must be one of the listed federations in the config.
 func (kd *KubeDNS) isFederationQuery(path []string) bool {
-	if len(path) == 4+len(kd.domainPath) &&
-		len(validation.IsDNS952Label(path[0])) == 0 &&
-		len(validation.IsDNS1123Label(path[1])) == 0 &&
-		len(validation.IsDNS1123Label(path[2])) == 0 &&
-		path[3] == serviceSubdomain {
-		for i, domComp := range kd.domainPath {
-			// kd.domainPath is reversed, so we need to look in the `path` in the reverse order.
-			if domComp != path[len(path)-i-1] {
-				return false
-			}
-		}
-		if _, ok := kd.federations[path[2]]; ok {
-			return true
+	if len(path) != 4+len(kd.domainPath) {
+		glog.Infof("len(%q) != 4+len(%q)", path, kd.domainPath)
+		return false
+	}
+	if errs := validation.IsDNS952Label(path[0]); len(errs) != 0 {
+		glog.Infof("%q is not an RFC 952 label: %q", path[0], errs)
+		return false
+	}
+	if errs := validation.IsDNS1123Label(path[1]); len(errs) != 0 {
+		glog.Infof("%q is not an RFC 1123 label: %q", path[1], errs)
```

Comment:
Would a better log message be: "not a federation query as %q is not an RFC 1123 label: %q"